tests/events/test_events_nyse.py

```diff
@@ -49,70 +44,108 @@ def test_edge_cases_for_TradingDayOfWeek(self,
+            # day after a holiday.
+            '2014-01-02': False,
+            # Test that the next Monday, which is at a start of a
+            # 'normal' week successfully triggers.
+            '2014-01-06': True,
+            # Test around a Monday holiday, MLK day, to exercise week
+            # start on a Tuesday.
+            # MLK is 2014-01-20 in 2014.
+            '2014-01-21': True,
+            # Should not trigger at market open of 01-22, a Wednesday.
+            '2014-01-22': False,
+        }
+
+        results = {
+            x: rule.should_trigger(self.cal.next_open(T(x)))
+            for x in expected.keys()
+        }
+
+        self.assertEquals(expected, results)
+
+        # Ensure that offset from start of week also works around edge cases.
+        rule = NthTradingDayOfWeek(1)
+        rule.cal = self.cal
 
-        n_triggered = 0
-        for m in jan_minutes:
-            if should_trigger(m):
-                self.assertEqual(m, next(trigger_minutes_iter))
-                n_triggered += 1
+        expected = {
+            # Should trigger at market open of 12-31, day after week start.
+            '2013-12-31': True,
+            # Should not trigger at market open of 1-2, a Thursday,
+            # day after a holiday.
+            '2014-01-02': False,
+            # Test around a Monday holiday, MLK day, to exercise
+            # week start on a Tuesday.
+            # MLK is 2014-01-20 in 2014.
+            # Should trigger at market open, two days after Monday hoilday.
+            '2014-01-22': True,
+            # Should not trigger at market open of 01-23, a Thursday.
+            '2014-01-23': False,
+        }
+
+        results = {
+            x: rule.should_trigger(self.cal.next_open(T(x)))
+            for x in expected.keys()
+        }
+
+        self.assertEquals(expected, results)
+
+        # `week_end`
+        rule = NDaysBeforeLastTradingDayOfWeek(0)
+        rule.cal = self.cal
 
-        self.assertEqual(n_triggered, expected_n_triggered)
+        expected = {
+            # Should trigger at market open of the Friday of the first week.
+            '2014-01-03': True,
+            # Should not trigger day before the end of the week.
+            '2014-01-02': False,
+            # Test around a Monday holiday, MLK day, to exercise week
+            # start on a Tuesday.
+            # MLK is 2014-01-20 in 2014.
+            # Should trigger at market open, on Friday after the holiday.
+            '2014-01-24': True,
+            # Should not trigger at market open of 01-23, a Thursday.
+            '2014-01-23': False,
+        }
+
+        results = {
+            x: rule.should_trigger(self.cal.next_open(T(x)))
+            for x in expected.keys()
+        }
+
+        self.assertEquals(expected, results)
 
     @parameterized.expand([('week_start',), ('week_end',)])
-    def test_week_and_time_composed_rule(self, type):
+    def test_week_and_time_composed_rule(self, taype):
```

Comment:
typo?

Reply:
This snuck in somehow while making the edit. Will fix.

However, `nosetests -x tests/events/test_events_nyse.py`  does pass even with the typo.
It is passing because we are using same name as the the built-in function `type` in this line, https://github.com/quantopian/zipline/blob/4cac769beebb22cadff84e42f93c37a1fcd2a554/tests/events/test_events_nyse.py#L116; resulting in always evaluating to the the `week_end` branch.